File: content/snippets/go/exercises/advanced/n_queens_problem.go
package main
import "fmt"
var N=8
var cols = make([]int, N)
func ok(r,c int) bool{ for i:=0;i<r;i++{ if cols[i]==c || abs(cols[i]-c)==r-i { return false } } return true }
func abs(x int) int{ if x<0 { return -x }; return x }
func solve(r int) bool{ if r==N { fmt.Println(cols); return true }; for c:=0;c<N;c++{ if ok(r,c){ cols[r]=c; if solve(r+1){ return true } } } return false }
func main(){ solve(0) }

File: content/snippets/python/exercises/advanced/n_queens_problem.py
N=8
cols=[0]*N

def ok(r,c):
    for i in range(r):
        if cols[i]==c or abs(cols[i]-c)==r-i: return False
    return True

def solve(r):
    if r==N:
        print(cols); return True
    for c in range(N):
        if ok(r,c):
            cols[r]=c
            if solve(r+1): return True
    return False

if __name__=='__main__': solve(0)
